File: основы17.09-2/main.py
import math

# 1
"""
x = float(input("Введите x: "))
y = float(input("Введите y: "))
f = float(input("Введите f: "))
n = 0
if ((x * y) > 0):
    n = ((f * (x) + y) ** 2) - math.sqrt(f * (x) * y)
elif ((x * y) < 0):
    n = ((f * (x) + y) ** 2) + (math.sqrt(math.log(f * (x) * y)))
elif (x * y == 0):
    n = ((f * (x) + y) ** 2) + 1

if n is None:
    print(f"Нет ответа")
elif n is not None:
    print(f"Ответ = {n}")
"""

# 2
"""
x = float(input("Введите x: "))
y = float(input("Введите y: "))
f = float(input("Введите f: "))
n = 0
if (x / y) > 0:
    n = math.log(f * (x)) + ((f * (x **2) + y) ** 3)
elif (x / y) < 0:
    n = math.log(abs(f * (x) / y)) + ((f * (x) + y) ** 3)
elif x == 0:
    n = (f * (x ** 2) + y) ** 3
elif y == 0:
    n = 0
if n is None:
    print(f"Нет ответа")
elif n is not None:
    print(f"Ответ = {n}")
"""

# 3
"""
x = float(input("Введите x: "))
a = float(input("Введите y: "))
f = float(input("Введите f: "))
n = 0

if 1 < abs(x) < 3:
    n = (((a * (x ** 2)) + 2) / ((x ** 2) + 1)) * (f * x)
elif abs(x) >= 3:
    n = (a ** 2) + (f * (x))
elif abs(x) <= 1:
    n = (a * x) * ((f * x) / (x + 2))
if n is None:
    print(f"Нет ответа")
elif n is not None:
    print(f"Ответ = {n}")
"""

# 4
"""
x = float(input("Введите x: "))
b = float(input("Введите b: "))
f = float(input("Введите f: "))
n = 0

if 1 < (x * b) < 3:
    n = math.exp(f * x)
elif 12 < (x * b) < 40:
    n = math.sqrt(abs((f * x) + (4 * b)))
else:
    n = (b * f * (x ** 2))
if n is None:
    print(f"Нет ответа")
elif n is not None:
    print(f"Ответ = {n}")
"""

# 5
x = float(input("Введите x: "))
y = float(input("Введите y: "))
z = float(input("Введите z: "))
f = float(input("Введите f: "))
m = 0

m = ((max(f*(x), y, z)) / (min(f*x, y))) + 5

if m is None:
    print(f"Нет ответа")
elif m is not None:
    print(f"Ответ = {m}")

File: основы17.09-2/main.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");

    // 1
    /*
    double x, y, f, n = 0;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите y: ";
    cin >> y;
    cout << "Введите f: ";
    cin >> f;

    if ((x * y) > 0) {
        n = ((f * (x) + y) * (f * (x) + y)) - sqrt(f * (x) * y);
    }
    else if ((x * y) < 0) {
        n = ((f * (x) + y) * (f * (x) + y)) + (sqrt(log(f * (x) * y)));
    }
    else if (x * y == 0) {
        n = ((f * (x) + y) * (f * (x) + y)) + 1;
    }

    if (n == 0 && false) {  // аналог твоего if n is None (у тебя n никогда None)
        cout << "Нет ответа" << endl;
    }
    else {
        cout << "Ответ = " << n << endl;
    }
    return 0;
    */

    // 2
    /*
    double x, y, f, n = 0;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите y: ";
    cin >> y;
    cout << "Введите f: ";
    cin >> f;

    if ((x / y) > 0) {
        n = log(f * (x)) + pow((f * (x * x) + y), 3);
    }
    else if ((x / y) < 0) {
        n = log(fabs(f * (x) / y)) + pow((f * (x) + y), 3);
    }
    else if (x == 0) {
        n = pow((f * (x * x) + y), 3);
    }
    else if (y == 0) {
        n = 0;
    }

    cout << "Ответ = " << n << endl;
    return 0;
    */

    // 3
    /*
    double x, a, f, n = 0;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите y: ";   // как у тебя в комментарии
    cin >> a;
    cout << "Введите f: ";
    cin >> f;

    if (1 < fabs(x) && fabs(x) < 3) {
        n = (((a * (x * x)) + 2) / ((x * x) + 1)) * (f * x);
    }
    else if (fabs(x) >= 3) {
        n = (a * a) + (f * (x));
    }
    else if (fabs(x) <= 1) {
        n = (a * x) * ((f * x) / (x + 2));
    }

    cout << "Ответ = " << n << endl;
    return 0;
    */

    // 4
    /*
    double x, b, f, n = 0;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите b: ";
    cin >> b;
    cout << "Введите f: ";
    cin >> f;

    if (1 < (x * b) && (x * b) < 3) {
        n = exp(f * x);
    }
    else if (12 < (x * b) && (x * b) < 40) {
        n = sqrt(fabs((f * x) + (4 * b)));
    }
    else {
        n = (b * f * (x * x));
    }

    cout << "Ответ = " << n << endl;
    return 0;
    */

    // 5
    /*
    double x, y, z, f, m = 0;

    cout << "Введите x: ";
    cin >> x;
    cout << "Введите y: ";
    cin >> y;
    cout << "Введите z: ";
    cin >> z;
    cout << "Введите f: ";
    cin >> f;

    m = ((max({f*(x), y, z})) / (min(f*x, y))) + 5;

    cout << "Ответ = " << m << endl;
    return 0;
    */
}